File: src/main/java/coza/trojanc/stonewriter/template/process/fields/ProcessedText.java
package coza.trojanc.stonewriter.template.process.fields;

import coza.trojanc.stonewriter.shared.Align;
import coza.trojanc.stonewriter.shared.Mode;

/**
 * Created by Charl-PC on 2016-10-11.
 */
public class ProcessedText{
	/** global printing mode bitmap - product of MODE_* constants above ORed together */
	private Mode mode = Mode.NORMAL;

	/** global alignment */
	private Align alignment = Align.LEFT;

	private String text;

	private Integer offset;

	public Mode getMode() {
		return mode;
	}

	public void setMode(Mode mode) {
		this.mode = mode;
	}

	public Align getAlignment() {
		return alignment;
	}

	public void setAlignment(Align alignment) {
		this.alignment = alignment;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public Integer getOffset() {
		return offset;
	}

	public void setOffset(Integer offset) {
		this.offset = offset;
	}
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/template/process/fields/ProcessedText.kt
package coza.trojanc.receipt.template.process.fields

import coza.trojanc.receipt.shared.Align
import coza.trojanc.receipt.shared.Mode

/**
 * @author Charl Thiem
 */
class ProcessedText {
    /** global printing mode bitmap - product of MODE_* constants above ORed together  */
    private var mode = Mode.NORMAL

    /** global alignment  */
    private var alignment = Align.LEFT

    private var text: String = "";

    private var offset: Int? = null

    fun getMode(): Mode {
        return mode
    }

    fun setMode(mode: Mode) {
        this.mode = mode
    }

    fun getAlignment(): Align {
        return alignment
    }

    fun setAlignment(alignment: Align) {
        this.alignment = alignment
    }

    fun getText(): String {
        return text
    }

    fun setText(text: String) {
        this.text = text
    }

    fun getOffset(): Int? {
        return offset
    }

    fun setOffset(offset: Int?) {
        this.offset = offset
    }
}
